docs/architecture/airbyte-specification.md

```diff
@@ -0,0 +1,180 @@
+                    "properties": {
+                      "name": {
+                        "type": "string"
+                      },
+                      "age": {
+                        "type": {
+                          "number"
+                        }
+                      }
+                    }
+                  }
+                }
+              ]
+            }
+            ```
+    * If you are using a data source that wraps an API with multiple different resources (e.g. `api/customers` and `api/products`) each route would correspond to a stream. The json object returned by each route would be described in the `json_schema` field.
+        * e.g. In the case where the API has two endpoints `api/customers` and `api/producs` and each returns a list of json objects, the `AirbyteCatalog` might look like this. (Note: using the json schema standard for defining a stream allows you to describe nested objects. You are not constrained to a classic "table/columns" structure)
+            ```
+            {
+              "streams": [
+                {
+                  "name": "customers",
+                  "schema": {
+                    "type": "object",
+                    "required": ["name"],
+                    "properties": {
+                      "name": {
+                        "type": "string"
+                      }
+                    }
+                  }
+                },
+                {
+                  "name": "products",
+                  "schema": {
+                    "type": "object",
+                    "required": ["name", "features"],
+                    "properties": {
+                      "name": {
+                        "type": "string"
+                      },
+                      "features": {
+                        "type": "array",
+                        "items": {
+                          "type": "object",
+                          "required": ["name", "productId"],
+                          "properties": {
+                            "name": "string",
+                            "productId": "number"
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              ]
+            }
+            ```
+#### Read
+* Input:
+    1. config: A configuration json object that has been validated using the `ConnectorSpecification`.
+    1. catalog: An `AirbyteCatalog`. This catalog should be a subset of the catalog returned by the `discover` command.
+    1. state: A json object. This object is only ever written or read by the source, so it is a json blob with whatever information is necessary to keep track of how much of the data source has already been read. Because, Airbyte currently only supports [Full Refresh](full-refresh.md), this state object is purely cosmetic. It will become more important when Airbyte beings to supporting incremental syncs.
```

Comment:
```suggestion
    1. state: A json object. This object is only ever written or read by the source, so it is a json blob with whatever information is necessary to keep track of how much of the data source has already been read. Because Airbyte currently only supports [Full Refresh](full-refresh.md), this state object is purely cosmetic. It will become more important when Airbyte supports incremental syncs.
```